src/main/java/com/microsoft/jenkins/azuread/AzureSecurityRealm.java

```diff
@@ -517,6 +520,14 @@ public SecurityComponents createSecurityComponents() {
                 } catch (GraphServiceException e) {
                     if (e.getResponseCode() == NOT_FOUND) {
                         return null;
+                    } else if (e.getResponseCode() == BAD_REQUEST) {
+                        LOGGER.log(Level.WARNING, "Failed to lookup user with userid '" + userId + "'."
+                                + " Enable 'Fine' Logging for more information.");
+                        StringWriter stacktraceStringWriter = new StringWriter();
+                        PrintWriter stacktracePrintWriter = new PrintWriter(stacktraceStringWriter);
+                        e.printStackTrace(stacktracePrintWriter);
+                        LOGGER.log(Level.FINE, stacktraceStringWriter.toString());
```

Comment:
it should be something like this:
```suggestion
                        LOGGER.log(Level.FINE, "Failed to lookup user with userid '" + userId, e);
```

Reply:
Didn't know the logger was capable of that, thanks. Tested it & it works.